src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -683,7 +696,7 @@ private static XmlSerializer[] GetSerializersFromCache(XmlMapping[] mappings, Ty
                 if (!s_xmlSerializerTable.TryGetValue(type, out typedMappingTable))
                 {
                     typedMappingTable = new Dictionary<XmlSerializerMappingKey, XmlSerializer>();
-                    s_xmlSerializerTable[type] = typedMappingTable;
+                    s_xmlSerializerTable.Add(type, typedMappingTable);
                 }
```

Comment:
Similar comment to elsewhere, should we have 2 backing stores, one for collectible and one for non-collectible? I don't know what the performance will be like on the TryGetValue.